### ISSUE TITLE
Shadowing error when field names matches package

### ISSUE THREAD

**Issue by dnaka91** [andrewhickman/protox#86]
I encountered a false positive shadowing error when the field name of a message matches the package name of an imported message. This is accepted by `protoc` but `protox` detects it as a shadowing issue.

### Minimal reproducible example

Having 2 Protobuf schemas with the following content:

In `other.proto`
```proto
syntax = "proto3";

package other;

message Other {}
```

In `sample.proto`
```proto
syntax = "proto3";

package sample;

import "other.proto";

message Sample {
  other.Other other = 1;
}
```

### Error message 
```sh
  Error:   × name 'other.Other' is shadowed
     ╭─[sample.proto:8:3]
   7 │ message Sample {
   8 │   other.Other other = 1;
     ·   ─────┬─────
     ·        ╰── found here
   9 │ }
     ╰────
    help: 'other.Other' is is resolved to 'sample.Sample.other.Other', which
          is not defined. The innermost scope is searched first in name
          resolution. Consider using a leading '.'(i.e., '.other.Other') to
          start from the outermost scope.

```

### Current workaround

When renaming the field the issue can be worked around, but changing the field names is not really an option in my bigger setup, as they are already being used in several places.

**Comment by andrewhickman** [comment]
Thanks for the report, I expect that you can work around this by locking the version of prost-reflect to version 0.14.0 which didn't have https://github.com/andrewhickman/prost-reflect/pull/126. I'll take a look at fixing this.

**Comment by andrewhickman** [comment]
This is also a re-occurrence of #57

**Comment by dnaka91** [comment]
Thanks for taking a look so quickly. I'll pin it to that version for now as a temporary workaround 👍.

**Comment by andrewhickman** [comment]
@dnaka91 Hopefully this is fixed in prost-reflect 0.14.2
